src/ui/component/errorBoundary/errorBoundary.tsx

```diff
@@ -1,26 +1,52 @@
+/* eslint-disable react-hooks/exhaustive-deps */
 /* eslint-disable react/destructuring-assignment */
-import { Component } from 'react'
-import type { ReactNode } from 'react'
+import { Component, useState, useEffect } from 'react'
+import type { SetStateAction, Dispatch, ReactNode } from 'react'
+import { useLocation } from 'react-router-dom'
 import { InternalError } from '@/ui/page/internalError/internalError'
 
-interface ErrorBoundaryProps {
-  children?: ReactNode
+type ErrorBoundaryInnerProps = {
+  children: JSX.Element
+  hasError: boolean
+  setHasError: Dispatch<SetStateAction<boolean>>
 }
-
-interface ErrorBoundaryState {
+type ErrorBoundaryProps = {
+  children: JSX.Element
+}
+type ErrorBoundaryInnerState = {
   hasError: boolean
 }
-
-export class ErrorBoundary extends Component<ErrorBoundaryProps, ErrorBoundaryState> {
-  public state: ErrorBoundaryState = {
-    hasError: false
+class ErrorBoundaryInner extends Component<ErrorBoundaryInnerProps, ErrorBoundaryInnerState> {
+  constructor(props: ErrorBoundaryInnerProps) {
+    super(props)
+    this.state = { hasError: false }
   }
-
-  public static getDerivedStateFromError(): ErrorBoundaryState {
+  static getDerivedStateFromError(): ErrorBoundaryInnerState {
     return { hasError: true }
   }
+  componentDidUpdate(prevProps: ErrorBoundaryInnerProps): void {
+    if (!this.props.hasError && prevProps.hasError) {
+      this.setState({ hasError: false })
+    }
+  }
+  componentDidCatch(): void {
+    this.props.setHasError(true)
+  }
 
-  public render(): ReactNode {
+  render(): ReactNode {
     return this.state.hasError ? <InternalError /> : this.props.children
   }
 }
+
+export const ErrorBoundary = ({ children }: ErrorBoundaryProps): JSX.Element => {
+  const [hasError, setHasError] = useState(false)
+  const location = useLocation()
+  useEffect(() => {
+    hasError && setHasError(false)
+  }, [location.key])
+  return (
```

Comment:
can you add a blank line above and below the `useEffect`please?

Reply:
Done!